libraries/AP_Math/AP_Math.h

```diff
@@ -82,7 +82,11 @@ inline bool is_negative(const T fVal1) {
  * @brief: Check whether a double is zero
  */
 inline bool is_zero(const double fVal1) {
-    return (fabsf(fVal1) < static_cast<double>(FLT_EPSILON));
+#ifdef ALLOW_DOUBLE_MATH_FUNCTIONS
+    return (fabs(fVal1) < DBL_EPSILON);
```

Comment:
this needs to use FLT_EPSILON, otherwise we end up with numbers that are not is_positiive(), not is_negative() and not is_zero()
it also could lead to some overflows when using a !is_zero() number in a single prec context

Reply:
I'm pretty sure that would not be semantically correct. There is a reason why FLT_EPSILON and DBL_EPSILON exist, to give a value "so close to zero that you can't tell the difference base on the inaccuracies of floating point math". Just using FLT_EPSILON when you have a double means that values that are semantically not zero will be "zero". How is that ok? the <double> version of is_positive() and is_negative() should use DBL_EPSILON. To avoid the problem you raised @tridge - perhaps is_zero() should be defined as !is_positive() && !is_negative().

Reply:
the thing is that we don't want "so close to zero that you can't tell the difference base on the inaccuracies of floating point math"
we're trying to avoid our controllers producing NaN and Inf values with complex equations. Going closer to the zero value with an equation that mixes single and double precision just makes it more likely we will blow up

Reply:
If the precision is not required/necessary or useful then perhaps then don't use double at all?